### PR TITLE
Fix complaints by -Wnull-dereference, enable in CI

### PR DESCRIPTION
**Pull request by dnnr**

### CODE REVIEW

File: src/mousedraghandler.cpp [hunk 1]
```diff
@@ -253,11 +253,18 @@ MouseResizeFrame::MouseResizeFrame(MonitorManager *monitors, shared_ptr<HSFrameL
...
+    if (!neighbour) {
         throw DragNotPossible("No neighbour frame in the direction of the cursor");
     }
+
     dragFrame_ = neighbour->getParent();
     auto df = dragFrame_.lock();
+
+    if (df == nullptr) {
+        // We need to check this explicitly because of -Wnull-dereference
+        throw DragNotPossible("No neighbour frame in the direction of the cursor");
```
Comment (blarz) [inline src/mousedraghandler.cpp:265]: That message isn't quite correct is it? It is the neighbours parent that is failing here?

File: src/frametree.cpp [hunk 2]
```diff
@@ -475,8 +475,10 @@ bool FrameTree::focusInDirection(Direction direction, bool externalOnly)
     shared_ptr<HSFrame> neighbour = curframe->neighbour(direction);
     if (neighbour) { // if neighbour was found
         shared_ptr<HSFrameSplit> parent = neighbour->getParent();
-        // alter focus (from 0 to 1, from 1 to 0)
-        parent->swapSelection();
+        if (parent) {
+            // alter focus (from 0 to 1, from 1 to 0)
+            parent->swapSelection();
+        }
         return true;
```
Comment (blarz) [inline src/frametree.cpp:482]: Is this correct even in the case where !parent? Or should the return true be inside the if(parent) block?

Comment (t-wissmann) [reply src/frametree.cpp:482]: If it has a neighbour, the neighbour must have a parent. So the only purpose of `if (parent)` is to silence the compiler warning.

File: src/mousedraghandler.cpp (hunk 1 above)
Comment (t-wissmann) [reply src/mousedraghandler.cpp:265]: This shouldn't happen anyway. The message is printed to the user by the `drag` command, so the only more helpful message is something like: `"The neighbour frame has no parent, this is a bug, please report"`.

File: src/frametree.cpp (hunk 2 above)
Comment (blarz) [reply src/frametree.cpp:482]: Ok, just wanted to be sure. Thanks for your explanation!

File: src/mousedraghandler.cpp (hunk 1 above)
Comment (dnnr) [reply src/mousedraghandler.cpp]: That makes sense, I've changed the message.

Comment (blarz) [reply src/mousedraghandler.cpp]: @dnnr just updated the message, resolving this.

**Comment by blarz** [comment]
> Looks good to me, this was the last -W exception, very nice!

Ah, it is not the last one (there is `no-sign-compare`), but one that bothered me quite much :D

**Pull request merged**